bpf/lib/encap.h

```diff
@@ -128,7 +129,12 @@ __encap_with_nodeid(struct __ctx_buff *ctx, __u32 tunnel_endpoint,
 		seclabel = LOCAL_NODE_ID;
 
 	node_id = bpf_htonl(tunnel_endpoint);
+#ifdef ENABLE_VTEP
+	if (vni == WORLD_ID)
+		key.tunnel_id = vni;
```

Comment:
What if the VNI field isn't `WORLD_ID`? Then `key.tunnel_id` won't get set, right? Is that intentional?

Reply:
Hi @joestringer this PR breaks existing identity tracing, are you trying to help me on the problem? I will get back to you :)

Reply:
Yeah I was just scanning through PRs that I haven't looked at in a while, and figured that if I see something, I should note it down even if we're not quite ready to address it yet :)

Reply:
I removed `if (vni == WORLD_ID)` since it looks redundant because I only pass `WORLD_ID` when VTEP_ENABLED, I tested again  still  this PR changed `identity 4352->4352` to `identity unknown->4352` for the ICMP reply when I ping from pod `10.0.1.34` with identity `4352` to remote pod `10.0.0.44`,  ping to vtep ip has the correct identity trace. I am thinking about make a vtep specific encap function so not to interfere with others, but the vtep specific encap function may have some overlap code with `__encap_with_nodeid()`,  before I do that, I will still try to see why this PR breaks the identity trace, @joestringer please let me know if you see where it breaks :)

Reply:
and above is what I have known it breaks identity trace, I am not sure if it breaks other traffic path identity trace too :), anyway I will dig more

Reply:
after tracing datapath debug log and VXLAN tcpdump, `unknown` identity means identity `0`, when ENABLE_VTEP,  all non-VTEP tunnel traffic pass get `key.tunnel_id` set to 0. so you are right on the spot, following fix the issue

```
#ifdef ENABLE_VTEP
        if (vni == WORLD_ID) {
                key.tunnel_id = vni;
        } else {
                key.tunnel_id = seclabel;
        }
#else
        key.tunnel_id = seclabel;
#endif
```